src/screens/PrivateChat/ContactsPicker/index.js

```diff
@@ -0,0 +1,253 @@
+import { connect } from 'react-redux';
+import {
+  View,
+  ScrollView,
+  Clipboard,
+} from 'react-native';
+import _ from 'lodash';
+import { Chip, Selectize } from 'react-native-material-selectize';
+import { addContact } from '../../../actions/contacts';
+import BackgroundImage from '../../../components/common/BackgroundImage';
+import styles from './styles';
+import FakeNavigationBar from '../../../components/common/FakeNavigationBar';
+import Loading from '../../../components/common/Loading';
+import NavigatorComponent from '../../../components/common/NavigatorComponent';
+import i18n from '../../../global/i18n';
+import Colors from '../../../global/colors';
+import type { Contact } from '../../../types/Contacts';
+import type { Navigator } from '../../../types/ReactNativeNavigation';
+import ScreenTitle from '../../../components/common/ScreenTitle';
+import AssetsImage from '../../../global/AssetsImages';
+import { imageSource } from '../../../utils/profile';
+import ListItem from '../../../components/common/ListItem';
+
+const DONE_BUTTON = 'DONE_BUTTON';
+const DISABLED_RIGHT_BUTTON = {
+  title: 'Done',
+  id: DONE_BUTTON,
+  buttonColor: Colors.navigationButtonColor,
+  disabled: true,
+};
+const ADD_CONTACT_ERROR = 'Error adding contact, please try again later.';
+
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc List of all contacts.
+   */
+  contacts: Array<Contact>,
+  /**
+   * @desc List of initially selected contacts.
+   */
+  initialSelectedContacts: Array<Contact>,
+  /**
+   * @desc Function to add a new contact
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  addContact: (identityKey: string, callback: (error: Error | null) => void) => void,
+  /**
+   * @desc Callback function to call when submitting contact selection.
+   */
+  onContactsSelected: (selectedContacts: Array<Contact>) => void,
+};
+
+type State = {
+  /**
+   * @desc The identityKey of the added contact.
+   */
+  addedContactIdentityKey: string,
+  /**
+   * @desc Flag whether loading is in progress.
+   */
+  loading: boolean,
+  /**
+   * @desc Error message for adding contact failure.
+   */
+  addContactError: string,
+  /**
+   * @desc Flag whether done button is disabled.
+   */
+  doneBtnDisabled: boolean,
+};
+
+class ContactsPickerScreen extends NavigatorComponent<Props, State> {
+  static navigatorButtons = {
+    leftButtons: [],
+    rightButtons: [DISABLED_RIGHT_BUTTON],
+  };
+
+  static defaultProps = {
+    initialSelectedContacts: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      addedContactIdentityKey: '',
+      loading: false,
+      addContactError: '',
+      doneBtnDisabled: true,
+    };
+  }
+
+  selectize: Selectize;
+
+  onNavBarButtonPress(id) {
+    if (id === DONE_BUTTON) {
+      const selectedContacts = this.selectize.getSelectedItems().result;
```

Comment:
I'm not sure if it's a good approach to let selectize component manage the selected items. Why don't we store selected items on the picker state? That would be more 'React-style', because using of components ref is discouraged unless it's the only possible solution. That also gives us an ability to remove `doneButtonDisabled` flag because it can be calculated from whether we have selected items. And we wouldn't have to manage done button enabling/disabling at all, because it will be just updated based on selected items field.